### PR TITLE
Holly remastered our unit tests and now they go hard

### PR DESCRIPTION
**Pull request by CarlosACJ55**
## Purpose

Added necessary unit tests to the database package such as TestNew, and formalized our TestMain and helper tests

## Usage

I also made sure to include the correct testing configuration "go test matcha" which makes it really clean to run all module tests

## Changes

- Moved tests into their own package fields to clear up a test logs a ton, also helped solve variable scope issues (i.e. we can now test private methods and attributes!!!)
- Reformatted go.yml and other files with issue
- Cleaned up PR template
- Commented out unnecessary tables for now since we want to keep them well tested and we have not gone through feature design meetings yet.
- Lastly, I changed the database field language from created_at to created_on to clarify that it holds a date not just a time
-I cleaned up every use of t.Error vs t.Fatal, please follow this pattern in the future.

## Rationale

This method of testing is far more reusable and actualy prevents anyone ushing breaking changes to our init.sql script and mysql.gop database implementation. It also allows us to test private helpers etc.

## Dependencies

N/A

## Issues and Bugs

- No password hashing yet
- Signup button still bugged, does not redirect to login page upon good signup

## Possible Solutions

N/A

## Additional Notes

N/A

## Contributors(@):

@CarlosACJ55

---

## Checklist before submitting

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have checked my code and corrected any misspellings.
- [x] I have confirmed that my PR is targeting the correct branch.
- [x] I understand that after merging, my branch may be deleted to keep the repository clean.
- [x] I have notified the team with the status of my PR.
- [x] If merge conflicts occur I will confer with other members of the team on how to proceed.

### CODE REVIEW

File: handlers.go [hunk 1]
```diff
@@ -99,6 +99,8 @@ func deleteUser(w http.ResponseWriter, r *http.Request) {
 		log.Println("Delete User failed -", err)
 		http.Redirect(w, r, "/settings", http.StatusSeeOther)
 	}
+
+	// TODO(@seoyoungcho213) : Remove user id cookie
```
Comment (Alishah634) [inline handlers.go:103]: What is this TODO update? is it still valid?

Comment (CarlosACJ55) [reply handlers.go:103]: Just a small change to remove the cookie from the ser's browser as a nicetude once they delete their user from our db

File: internal/database/mysql_test.go [hunk 2]
```diff
@@ -38,78 +37,134 @@ func setupDBAndOpenSubject(t *testing.T) *internalDatabase.MySQLDatabase {
...
-	return subject
+func TestConnections(t *testing.T) {
+	subject, probe := setup(t)
+	if err := subject.underlyingDB.Ping(); err != nil {
+		t.Fatal("Failed to open subject db properly -", err)
+	}
+	if err := probe.Ping(); err != nil {
+		t.Fatal("Failed to open probe db properly -", err)
+	}
+	teardown(t, subject, probe)
```
Comment (Alishah634) [inline internal/database/mysql_test.go:76]: This Ping() and the one after it, are already performed in the teardown function. Is it still necessary to do this?

File: internal/database/mysql_test.go [hunk 3]
```diff
@@ -38,78 +37,134 @@ func setupDBAndOpenSubject(t *testing.T) *internalDatabase.MySQLDatabase {
...
+				if len(expectedCols) != 0 {
+					t.Log("Failed to create columns:")
+					for k := range expectedCols {
+						t.Log(k)
+					}
+					t.Fail()
+				}
+			},
+		)
+		delete(expectedTables, table)
```
Comment (Alishah634) [inline internal/database/mysql_test.go:130]: Lines 112 to 138, is there a more efficient solution?

**Review by Alishah634 — APPROVED** [review]
Overall, no major issues, comments have been left regarding areas of concern please do not merge/rebase till you are certain they will have no consequences. Great work!

**Review by Alishah634 — APPROVED** [review]

File: internal/database/mysql_test.go (hunk 2 above)
Comment (CarlosACJ55) [reply internal/database/mysql_test.go:76]: I am really only leaving TestConnections for completeness, given how our setup and teardown functions are currently written, this test is really redundant. We have agreed to leave for now tho.

File: internal/database/mysql_test.go (hunk 3 above)
Comment (CarlosACJ55) [reply internal/database/mysql_test.go:130]: There is maybe a more readable solution, but any other sort of checking woudl require extra memory/ iterative checking at the end, so not really.

**Pull request merged**